ui/pages/confirm-transaction/confirm-transaction.component.js

```diff
@@ -159,7 +159,12 @@ export default class ConfirmTransaction extends Component {
       transactionId &&
       (!paramsTransactionId || paramsTransactionId === transactionId);
 
-    if (isTokenMethodAction && validTransactionId) {
+    const value = transaction?.txParams?.value;
+    if (
+      isTokenMethodAction &&
+      validTransactionId &&
+      (!value || value === '0x0')
```

Comment:
I am wondering, why check for `(!value || value === '0x0')` rather than `isSendWithApproval` here?

Reply:
Updated